creator/refferal_tokens/schema.py

```diff
@@ -0,0 +1,146 @@
+            "role": ["exact"],
+            "study": ["exact"],
+            "group": ["exact"],
+            "created_at": ["gt", "lt"],
+        }
+
+    order_by = OrderingFilter(fields=("created_at",))
+
+
+class CreateReferralTokenMutation(graphene.Mutation):
+    class Arguments:
+        email = graphene.String(
+            required=True,
+            description="The email address to send the referral to",
+        )
+
+        study_id = workflows = graphene.List(
+            graphene.ID,
+            description="A list of studies that the user will be added to upon exchange of the token",
+            required=False,
+        )
+
+        groups = graphene.List(
+            graphene.ID,
+            required=False,
+            description="The permission group the user will be given after exchanging the token",
+        )
+
+        role = graphene.String(
+            required=True,
+            description="The collaborator role the user will be given in each of the studies",
+        )
+
+    referral_token = Field(ReferralTokenNode)
+
+    def mutate(self, info, input):
+        """
+        Create a new referral token with a given workflow type and study if that study
+        does not already have a project with that workflow type.
+        """
+
+        user = info.context.user
+
+        if not user.has_perm("referralTokens.add_referralToken"):
+            raise GraphQLError("Not allowed")
+
+        try:
+            _, kf_id = from_global_id(input["study"])
+            study = Study.objects.get(kf_id=kf_id)
+        except Study.DoesNotExist:
+            raise GraphQLError("Study does not exist.")
+
+        try:
+            referral_token = ReferralToken(
+                email=email, role=role, study=study, groups=groups,
+            )
+            version.save()
+        except ClientError:
+            raise GraphQLError("Failed to save referral token")
+
+        # Log an event
+        if user:
+            message = (
+                f"{user.username} created referral token {referral_token.id}"
+            )
+        else:
+            message = f"A new referral token was created {referral_token.id}"
+        event = Event(study=study, description=message, event_type="RT_CRE",)
+
+        return CreateReferralTokenMutation(referral_token=referral_token)
+
+
+class Query(object):
+    referral_token = relay.Node.Field(
+        ReferralTokenNode, description="Get a single referral token"
+    )
+    all_referral_tokens = DjangoFilterConnectionField(
+        ReferralTokenNode,
+        filterset_class=ReferralTokenFilter,
+        description="Get all referral tokens",
+    )
+
+    def resolve_all_referral_tokens(self, info, **kwargs):
+        """
+        Return all referral tokens if user has view_all_referral_tokens
+        Return not allowed otherwise
+        """
+        user = info.context.user
+
+        if not (user.has_perm("referralTokens.view_all_referral_tokens")):
```

Comment:
Should be `referral_tokens.view_all_referral_tokens` and we should add the `view_all_referral_tokens` permission to the model.